GitUI/CommandsDialogs/SettingsDialog/Pages/ShellExtensionSettingsPage.cs

```diff
@@ -46,6 +66,17 @@ protected override void PageToSettings()
 
             AppSettings.CascadeShellMenuItems = l_CascadeShellMenuItems;
             AppSettings.AlwaysShowAllCommands = cbAlwaysShowAllCommands.Checked;
+            if (!ShellExtensionManager.IsNotInstalled() && cbEnableIntegration.Checked == ShellExtensionManager.IsNotRegistered())
+            {
+                if (cbEnableIntegration.Checked)
+                {
+                    ShellExtensionManager.RegisterShellExtensionSafe();
+                }
+                else
+                {
+                    ShellExtensionManager.UnregisterShellExtensionSafe();
+                }
+            }
```

Comment:
What are we doing here?

Reply:
Do nothing in following cases:
1. Binaries weren't installed
2. Checkbox and shell registration status match each other
I should fix naming IsNotRegistered => IsRegistered so it will be more clear